src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHostRawUserInterface.cs

```diff
@@ -1745,10 +1746,10 @@ public override string ToString()
         /// from the keyboard device, blocking processing until a keystroke is
         /// typed that matches the specified keystroke options.
         /// </summary>
-        /// <param name="options">Unused</param>
+        /// <param name="options">Only NoEcho is supported.</param>
         public override KeyInfo ReadKey(ReadKeyOptions options)
         {
-            ConsoleKeyInfo key = Console.ReadKey();
+            ConsoleKeyInfo key = Console.ReadKey(options.HasFlag(ReadKeyOptions.NoEcho));;
```

Comment:
Please avoid the use of the `HasFlag` api - it is much, much slower than the simple bit test - see the implementation [here](https://github.com/dotnet/coreclr/blob/13e7c4368da664a8b50228b1a5ef01a660fbb2dd/src/vm/reflectioninvocation.cpp#L3064). The expensive part is mostly boxing, but you can see it does other work before the simple bitwise test.